### PR TITLE
Handle unsupported date formats for DateFilter

### PR DESCRIPTION
**Pull request by anvuong119**
JIRA: SD-1250

---

Supported PR commands:

| Command         | Description            |
| --------------- | ---------------------- |
| `ok to test`    | Re-run standard checks |
| `extended test` | BackstopJS tests       |

---

# PR Checklist

-   [x] commit messages adhere to the [commit message guidelines](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#what-should-the-commits-look-like)
-   [x] `check` passes
-   [ ] `check-extended` passes
-   [x] `rush change` [was run if applicable](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#how-do-i-describe-my-changes-for-the-changelog)

### CODE REVIEW

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/1660/ : SUCCESS in 4s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/3234/ : SUCCESS in 13m 46s

**Comment by anvuong119** [comment]
extended test

**Comment by gdgate** [comment]
Build succeeded (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/1661/ : SUCCESS in 7s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-backstop-tests/523/ : SUCCESS in 31m 30s

File: libs/sdk-ui-filters/src/DateFilter/DateFilterCore.tsx [hunk 1]
```diff
@@ -54,6 +56,21 @@ const DropdownBody: React.FC<{
...
 
+export const verifyDateFormat = (dateFormat: string): string => {
+    try {
+        // Try to format the current date to verify if dateFormat is a valid format.
+        formatAbsoluteDate(new Date(), dateFormat);
+        return dateFormat;
+    } catch {
+        // If an error occurs, then dateFormat is invalid and the default format should be used instead. Also, a warning is written in the console.
+        // eslint-disable-next-line no-console
+        console.log(
```
Comment (no23reason) [inline libs/sdk-ui-filters/src/DateFilter/DateFilterCore.tsx:67]: Please make this a console.warn so that it is easier to spot.

**Review by no23reason — APPROVED** [review]
LGTM

**Comment by gdgate** [comment]
Build in pipeline check aborted.

**Comment by gdgate** [comment]
Build succeeded (gate pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-fossa-scanning-zuul-el7/524/ : SUCCESS in 29s (non-voting)
- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-merge-compliance-check/784/ : SUCCESS in 20s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/3238/ : SUCCESS in 13m 39s

**Pull request merged**